modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_lp_timer_zephyr.c

```diff
@@ -18,60 +18,35 @@
 static volatile bool m_clock_ready;
 static bool m_is_running;
 static int32_t m_rtc_channel;
-static bool m_in_critical_section;
+static uint32_t m_in_critical_section;
+static uint64_t m_last_scheduled_time;
 
-void rtc_irq_handler(int32_t id, uint32_t cc_value, void *user_data)
+void rtc_irq_handler(int32_t id, uint64_t target_time, void *user_data)
 {
-	(void)cc_value;
 	(void)user_data;
-	nrf_802154_sl_mcu_critical_state_t state;
 
 	assert(id == m_rtc_channel);
 
-	nrf_802154_sl_mcu_critical_enter(state);
+	bool was_running;
 
+	was_running = m_is_running;
 	m_is_running = false;
-	(void)z_nrf_rtc_timer_compare_int_lock(m_rtc_channel);
-
-	nrf_802154_sl_mcu_critical_exit(state);
-
-	nrf_802154_lp_timer_fired();
-}
-
-/**
- * @brief Start one-shot timer that expires at specified time on desired channel.
- *
- * Start one-shot timer that will expire @p dt microseconds after @p t0 time on channel @p channel.
- *
- * @param[in]  channel  Compare channel on which timer will be started.
- * @param[in]  t0       Number of microseconds representing timer start time.
- * @param[in]  dt       Time of timer expiration as time elapsed from @p t0 [us].
- */
-static void timer_start_at(uint32_t channel,
-			   uint32_t t0,
-			   uint32_t dt)
-{
-	uint32_t cc_value = NRF_802154_SL_US_TO_RTC_TICKS(t0 + dt);
-	nrf_802154_sl_mcu_critical_state_t state;
-
-	z_nrf_rtc_timer_compare_set(m_rtc_channel, cc_value, rtc_irq_handler, NULL);
-
-	nrf_802154_sl_mcu_critical_enter(state);
 
-	m_is_running = true;
-	z_nrf_rtc_timer_compare_int_unlock(m_rtc_channel, (m_in_critical_section == false));
+	__DMB();
 
-	nrf_802154_sl_mcu_critical_exit(state);
+	if (was_running) {
+		nrf_802154_sl_timer_handler(target_time);
```

Comment:
Won't this create a possiblilty, where a compare match event is lost? I think `nrf_802154_sl_timer_handler` permits spurious interrupt.